CharityPortal/api/middlewares/session.go

```diff
@@ -7,42 +7,28 @@ import (
 
 type SessionMiddleware struct {
 	authProvider   auth.AuthProvider
-	appEnvironment string
 }
 
-func NewSessionMiddleware(authProvicer auth.AuthProvider, appEnvironment string) *SessionMiddleware {
+func NewSessionMiddleware(authProvicer auth.AuthProvider) *SessionMiddleware {
 	return &SessionMiddleware{
 		authProvider:   authProvicer,
-		appEnvironment: appEnvironment,
 	}
 }
 
 func (am *SessionMiddleware) Session(h http.Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		if am.appEnvironment == "development" {
-			// In development mode, we skip session management for simplicity.
-			sessionData := auth.UserClaims{
-				Name:  "test",
-				Email: "test@test.com",
-				Sub:   "test-sub",
-			}
-			r = r.WithContext(auth.SetUserContext(r.Context(), &sessionData))
-			h.ServeHTTP(w, r)
-			return
+		var cookieVal string
+		if cookie, err := r.Cookie("session"); err == nil {
+			cookieVal = cookie.Value
 		}
-		cookie, err := r.Cookie("session")
-		if err != nil {
-			h.ServeHTTP(w, r)
+		sessionData, err := am.authProvider.Decode("session", cookieVal)
+		if sessionData == nil {
+			clearSessionCookie(w)
+			http.Error(w, "Unauthorized", http.StatusUnauthorized)
```

Comment:
In session middleware i just wanted to set session if exists, if not then bypass request to next auth middleware which will determine whether you are authorized or not.
Now, production doesn't work since we are returning Unauthorized for `/login` path which doesn't have session cookie.